checker/bin/mapreduce-checker.sh

```diff
@@ -0,0 +1,121 @@
+#
+# This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+# either express or implied, as more fully set forth in the License.
+#
+# See the NOTICE file distributed with this work for information regarding copyright ownership.
+#
+
+LAUNCHER=
+# If debugging is enabled propagate that through to sub-shells
+if [[ "$-" == *x* ]]; then
+  LAUNCHER="bash -x"
+fi
+BIN=$(cd "$( dirname "$( readlink "$0" || echo "$0" )" )"; pwd)
+
+USAGE="Usage: alluxio-checker.sh mapreduce [NUM_MAPS]
+where NUM_MAPS is an optional argument which affects the number of map tasks in MapReduce job of integration checker.
+  We recommend users to set this number according to your Hadoop cluster size,
+  so that MapReduce integration checker can check more Hadoop nodes.
+  By default, the INPUT_SPLITS is 10.
+
+-h  display this help."
+
+ALLUXIO_PATH=$(cd "${BIN}/../../"; pwd)
+ALLUXIO_JAR_PATH=""
+HADOOP_LOCATION=""
+NUM_MAPS="${1:-10}";
+
+# Find the location of hadoop in order to trigger Hadoop job
+function find_hadoop_path() {
+  # If ${HADOOP_HOME} has been set
+  if [[ "${HADOOP_HOME}" != "" ]]; then
+    {
+      [ -f "${HADOOP_HOME}/hadoop" ] && HADOOP_LOCATION="${HADOOP_HOME}/hadoop"
+    } ||   {
+      [ -f "${HADOOP_HOME}/bin/hadoop" ] && HADOOP_LOCATION="${HADOOP_HOME}/bin/hadoop"
+    } || {
+      [ -f "${HADOOP_HOME}/libexec/bin/hadoop" ] && HADOOP_LOCATION="${HADOOP_HOME}/libexec/bin/hadoop"
+    } || { # Try to find hadoop executable file in HADOOP_HOME
+      if [[ "${HADOOP_LOCATION}" == "" ]]; then
+        array=(`find "${HADOOP_HOME}" -type f -name 'hadoop'`)
+        for i in "${array[@]}"; do
+          HADOOP_LOCATION="$i"
+          break;
+        done
+      fi
+    }
+  fi
+
+  # If ${PATH} has been set
+  if [[ "${HADOOP_LOCATION}" == "" ]]  &&  [[ "${PATH}" != "" ]]; then
+    IFS=':' read -ra PATHARR <<< "$PATH"
+    for p in "${PATHARR[@]}"; do
+      if [ -f "$p/hadoop" ]; then
+        HADOOP_LOCATION="$p"/hadoop
+        break;
+      fi
+    done
+  fi
+
+  if [[ "${HADOOP_LOCATION}" == "" ]]; then
+    if [[ "${HADOOP_HOME}" != "" ]]; then
+      echo -e "Cannot find executable file hadoop in your HADOOP_HOME: ${HADOOP_HOME}, please check your HADOOP_HOME." >&2
+    else
+      echo -e "Please set HADOOP_HOME before running MapReduce integration checker." >&2
+    fi
+    exit 1
+  fi
+}
+
+function trigger_mapreduce() {
+  # Without -libjars, we assume that the Alluxio client jar has already been distributed on the classpath of all Hadoop nodes
+  ${LAUNCHER} "${HADOOP_LOCATION}" jar "${BIN}/../target/alluxio-checker-${VERSION}-jar-with-dependencies.jar" \
+    alluxio.checker.MapReduceIntegrationChecker "${NUM_MAPS}"
+
+  # Use -libjars if the previous attempt failed because of unable to find Alluxio classes and add remind information
+  if [[ "$?" == 2 ]]; then
+    ${LAUNCHER} "${HADOOP_LOCATION}" jar "${BIN}/../target/alluxio-checker-${VERSION}-jar-with-dependencies.jar" \
```

Comment:
I found that to get the checker to work locally, I needed to add `export HADOOP_CLASSPATH=/path/to/alluxio/jar` before this